### PR TITLE
Stabilize comments on ObjectTypeAnnotation

### LINKED ISSUE #1211: Unstable comment in flow object
I kept the original comment as it is funny :)

```js
type Props = // (DispatchProps & StateProps); WHY DON'T YOU WORK FLOW!!!!!!!!!
{
  isPlaying: boolean,
};
```

```js
type Props = { // (DispatchProps & StateProps); WHY DON'T YOU WORK FLOW!!!!!!!!!
  isPlaying: boolean
};
```

```js
type Props = {
  // (DispatchProps & StateProps); WHY DON'T YOU WORK FLOW!!!!!!!!!
  isPlaying: boolean
};
```

https://prettier.github.io/prettier/#%7B%22content%22%3A%22type%20Props%20%3D%20%2F%2F%20(DispatchProps%20%26%20StateProps)%3B%20WHY%20DON'T%20YOU%20WORK%20FLOW!!!!!!!!!%5Cn%7B%5Cn%20%20isPlaying%3A%20boolean%2C%5Cn%7D%3B%5Cn%5Cntype%20Props%20%3D%20%7B%20%2F%2F%20(DispatchProps%20%26%20StateProps)%3B%20WHY%20DON'T%20YOU%20WORK%20FLOW!!!!!!!!!%5Cn%20%20isPlaying%3A%20boolean%5Cn%7D%3B%5Cn%5Cntype%20Props%20%3D%20%7B%5Cn%20%20%2F%2F%20(DispatchProps%20%26%20StateProps)%3B%20WHY%20DON'T%20YOU%20WORK%20FLOW!!!!!!!!!%5Cn%20%20isPlaying%3A%20boolean%5Cn%7D%3B%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22%22%2C%22doc%22%3Afalse%7D%7D

### PR DESCRIPTION
**Pull request by existentialism**
Fixes #1211.

### CODE REVIEW

File: tests/flow_comments/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -29,3 +29,69 @@ const beep = (data /*: Object*/) => {};
...
+  isPlaying: boolean
+};
+
+type Props = {
+  // (DispatchProps & StateProps); WHY DON'T YOU WORK FLOW!!!!!!!!!
+  isPlaying: boolean
+};
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+type Props =
+  // (DispatchProps & StateProps); WHY DON'T YOU WORK FLOW!!!!!!!!!
```
Comment (vjeux) [inline tests/flow_comments/__snapshots__/jsfmt.spec.js.snap]: I don't think we want that

File: src/comments.js [hunk 2]
```diff
@@ -734,6 +735,19 @@ function handleClassMethodComments(enclosingNode, comment) {
   return false;
 }
 
+function handleTypeAliasComments(enclosingNode, followingNode, comment) {
+  if (
+    enclosingNode &&
+    enclosingNode.type === "TypeAlias" &&
+    followingNode &&
+    followingNode.type === "ObjectTypeAnnotation"
```
Comment (vjeux) [inline src/comments.js]: It's likely going to be an issue with all the types, not only objects.

What about if `enclosingNode === "TypeAlias"` then put the comment as leading of the enclosingNode?

File: tests/flow_comments/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (existentialism) [inline tests/flow_comments/__snapshots__/jsfmt.spec.js.snap]: Ok, I was originally matching it how the `ObjectExpression` version of this behaves, but I guess:

```js
// (DispatchProps & StateProps); WHY DON'T YOU WORK FLOW!!!!!!!!!
type Props = {
  isPlaying: boolean
};
```

Is desired output?

File: src/comments.js (hunk 2 above)
Comment (existentialism) [inline src/comments.js]: Results in:

```js
// (DispatchProps & StateProps); WHY DON'T YOU WORK FLOW!!!!!!!!!
type Props = {
  isPlaying: boolean
};
```

Probably want to stabilize as:

```js
type Props = {
  // (DispatchProps & StateProps); WHY DON'T YOU WORK FLOW!!!!!!!!!
  isPlaying: boolean
};
```

yeah?

Comment (vjeux) [inline src/comments.js]: That looks good to me

**Comment by vjeux** [comment]
Thanks!

**Pull request merged**